src/gc/proxy.d

```diff
@@ -318,6 +321,20 @@ extern (C)
         return proxy.gc_inFinalizer();
     }
 
+    void gc_lock() nothrow @nogc
+    {
+        if( proxy is null )
+            return _gc.lock();
+        return proxy.gc_lock();
+    }
+
+    void gc_unlock() nothrow @nogc
+    {
+        if( proxy is null )
+            return _gc.unlock();
+        return proxy.gc_unlock();
+    }
```

Comment:
Hmm...

I thought this was used exclusively for connecting multiple Druntime instances (from multiple shared objects). Is this also used for third-party GC implementations? ( @rainers , @leandro-lucarella-sociomantic )?

If it is only used for the former, then it should be a non-issue, because AFAIK we don't consider the Druntime ABI binary-stable, and mixing different Druntime versions in one application is unsupported.

Still, if I understand the intended use of this proxy thing correctly, this code does need to be here. When we are in a shared object and forking the program, we do need to acquire the exclusive lock of the main program's GC instance.